src/comm/TcpAcceptor.cc

```diff
@@ -302,6 +302,8 @@ Comm::TcpAcceptor::acceptOne()
         return;
     }
 
+    newConnDetails->nfmark = Ip::Qos::getNfmarkFromConnection(newConnDetails, Ip::Qos::Incoming);
```

Comment:
This adds performance overhead for all accepted connections even if the admin is not using the new ACL, right? How expensive are the associated system calls? Would it be easy to bypass this assignment when the new ACL is not used?

Similar concern applies to the FwdState code, but we were extracting the possibly useless mark there before your changes so I cannot argue a performance regression there.